Router.svelte

```diff
@@ -157,15 +157,36 @@ export function link(node) {
     // Destination must start with '/'
     const href = node.getAttribute('href')
     if (!href || href.length < 1 || href.charAt(0) != '/') {
-        throw Error('Invalid value for "href" attribute')
+        throw Error('Invalid value for "href" attribute: ' + href)
     }
 
     // Add # to every href attribute
     node.setAttribute('href', '#' + href)
+
+    // If a user wants to restore scroll positions, we do this to know when they are about to navigate to a new 'page'
+    // TODO: it seems I don't have reference to this var from context='module'; will need to fix this
+    //if (restoreScrollPosition) {
+    node.addEventListener('click', scrollstateHistoryHandler)
```

Comment:
Ok, few things here:

1. I am wondering if this would be problematic? See #39 and why we changed the "link" action to update the href attribute rather than just listening on the click event
2. I am not a big fan of the fact that this would make restoreScrollPosition work only when links are created with use:link. I think that giving developers the ability to navigate with normal links (`<a href="#/something">`) is powerful, as well as being able to use `push` and `pop`
3. In any case, no need to add an `if (restoreScrollPosition)` here. It's totally fine if the listener is listening in any case

Reply:
1. Do you mean #33? If so, I just tested and middle click behaves as expected. Agree that doing something reactive like the href will be better, but in this case we need to be a level higher to know if the user clicks on something (forward navigation). 
2. True, haven't gotten to the rest yet. Do you mean you'd want to provide restoreScrollPosition functionality even on links not created with `use:link`? That can be done. And `push` and `pop` can be modified too to support it, i'll do that.
3. Cool

Reply:
Yes sorry I meant 33. What about "right click - open"? (But I guess that maybe doesn't matter at all, since it wouldn't open in the same browser view anyways?)

And yes, if you implement something it needs to work on all links, not just those using `use:link` :)

Reply:
That works, and yes it doesn't seem to break the current page's history.

Yess. See updated implementation. Interesting note that using a lambda caused things to slow way down after a few page loads due to presumably mem leak.